cmd/juju/controller/destroy.go

```diff
@@ -250,7 +250,8 @@ upgrade the controller to version 2.3 or greater.
 		}
 
 		updateStatus := newTimedStatusUpdater(ctx, api, controllerEnviron.Config().UUID(), clock.WallClock)
-		envStatus := updateStatus(0)
+		// wait for 2 seconds to let empty hosted models changed from alive to dying.
+		envStatus := updateStatus(2 * time.Second)
```

Comment:
We can't do this - we can't rely on an arbitrary wait time. It is fragile and will break on systems like prodstack that are heavily loaded. Without the wait time, what happened? An error? It may be that we need to introduce a new for loop to poll status to wait for models to transition to dead